evm/hyperliquid.js

```diff
@@ -0,0 +1,225 @@
+    printInfo('Token address', tokenAddress);
+
+    try {
+        const TokenContract = getContractJSON('HyperliquidInterchainToken');
+        const token = new Contract(tokenAddress, TokenContract.abi, wallet);
+
+        const currentDeployer = await token.deployer();
+        printInfo('Current deployer', currentDeployer);
+    } catch (error) {
+        throw error;
+    }
+}
+
+async function updateTokenDeployer(wallet, chain, options) {
+    const [tokenId, deployer] = options.args;
+    validateParameters({
+        isNonEmptyString: { tokenId },
+        isValidAddress: { deployer },
+    });
+
+    const contracts = chain.contracts;
+    const interchainTokenFactoryAddress = contracts.InterchainTokenFactory?.address;
+    const interchainTokenServiceAddress = contracts.InterchainTokenService?.address;
+
+    validateParameters({
+        isValidAddress: { interchainTokenFactoryAddress, interchainTokenServiceAddress },
+    });
+
+    const IInterchainTokenService = getContractJSON('IInterchainTokenService');
+    const interchainTokenService = new Contract(interchainTokenServiceAddress, IInterchainTokenService.abi, wallet);
+
+    const tokenAddress = await interchainTokenService.registeredTokenAddress(tokenId);
+    printInfo('Token address', tokenAddress);
+
+    const ServiceContract = getContractJSON('HyperliquidInterchainTokenService');
+    const service = new Contract(interchainTokenServiceAddress, ServiceContract.abi, wallet);
+
+    const TokenContract = getContractJSON('HyperliquidInterchainToken');
+    const token = new Contract(tokenAddress, TokenContract.abi, wallet);
+
+    const currentDeployer = await token.deployer();
+    printInfo('Current deployer', currentDeployer);
+    printInfo('New deployer', deployer);
+
+    const serviceOwner = await service.owner();
+    const isOperator = await service.isOperator(wallet.address);
+
+    if (wallet.address.toLowerCase() !== serviceOwner.toLowerCase() && !isOperator) {
+        throw new Error('Wallet does not have permission to update deployers. Must be service owner or operator.');
+    }
+
+    const gasOptions = await getGasOptions(chain, options, 'HyperliquidInterchainTokenService');
+    const tx = await service.updateTokenDeployer(tokenId, deployer, gasOptions);
+    await handleTx(tx, chain, service, 'updateTokenDeployer');
+
+    const updatedDeployer = await token.deployer();
+    printInfo('Updated deployer', updatedDeployer);
+}
+
+async function main(processor, args, options) {
+    options.args = args;
+    return mainProcessor(options, (config, chain, options) => {
+        if (!isHyperliquidChain(chain)) {
+            throw new Error(`Chain "${chain.name}" is not supported. This script only works on Hyperliquid chains.`);
+        }
+
+        const rpc = chain.rpc;
+        const provider = getDefaultProvider(rpc);
+        const wallet = new Wallet(options.privateKey, provider);
+
+        return processor(wallet, chain, options);
+    });
```

Comment:
```suggestion
    return mainProcessor(options, (config, chain, options) => {
        if (!isHyperliquidChain(chain)) {
            throw new Error(`Chain "${chain.name}" is not supported. This script only works on Hyperliquid chains.`);
        }

        const rpc = chain.rpc;
        const provider = getDefaultProvider(rpc);
        const wallet = new Wallet(options.privateKey, provider);

        return processor(wallet, chain, args, options);
    });
```
just pass `args` directly since you're using a closure anyways